mods/gen2/moves.js

```diff
@@ -12,13 +12,17 @@ exports.BattleMovedex = {
 			if (target.boosts.atk >= 6) {
 				return false;
 			}
+
+			var boost = 0;
+
 			if (target.hp <= target.maxhp / 2) {
-				this.boost({atk: 2});
-				return false;
+				boost = (target.boost ? target.boost.atk : 0) + 2;
```

Comment:
Actually, I think I figured it out. It's because in `Battle.prototype.boost`, it's set to not display any messages when Belly Drum is used (this was to resolve issues with Contrary Belly Drum in the latest generation). That needs to be fixed to display the message if `this.gen < 6` or something.

Basically, it's silently boosting and your mon will have +2 Attack, but it doesn't say it so the client never gets the memo. Then, the `return false` makes it think the move failed.